### PR TITLE
[1124] - Use Mempool For Fee Rate

### LINKED ISSUE #1124: Mining Fees in Header Stuck at 116 sat/vB
**Description**
The approximate mining fees in the header carousel appears to be stuck at 116 sat/vB

**Steps to Reproduce**
1. View page header
2. Click on carousel until you land on approximate fees

**Expected behavior**
The mining fees used to follow reality pretty well. But for some reason, it's stuck now.

**Screenshots**
image

**Logs**
N/A

**Environment:**
N/A

**Additional context**
Discussed [here](https://stacker.news/items/522551/r/WeAreAllSatoshi?commentId=522975) and [here](https://stacker.news/items/521675/r/WeAreAllSatoshi?commentId=522401)

@ekzyis suggested maybe switching to use mempool.space's API instead of whatever we're doing (querying our LND node, I believe?)

It remains to be determined if the issue is with LND, or something in our app layer itself.

### PR DESCRIPTION
**Pull request by tsmith123**
Fixes #1124 

## Description
This PR addresses an issue with the transaction fee rate shown in the header as it wasn't updating regularly.
- Uses Mempool for latest transaction fee (uses `hourFee` but can be changed to higher or lower priority if requested)
- Tidies up onClick handler to reduce repetitive code

**Note:** the existing implementation deletes the local storage entry on every complete cycle of clicking the price component. However, with these changes nothing is deleted as it doesn't appear necessary. Please let me know if I've overlooked something.

Tested locally on desktop using a minimal compose profiles setup. Might benefit from some extra eyes on this as it's my first PR.

### CODE REVIEW

File: api/resolvers/chainFee.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-  }
+  const url = 'https://mempool.space/api/v1/fees/recommended'
+  const chainFee = await fetch(url)
+    .then((res) => res.json())
+    .then((body) => body.hourFee)
+    .catch((err) => {
+      console.error('fetchChainFee', err)
+      return 0
+    })
+
```
Comment (tsmith123) [inline api/resolvers/chainFee.js:13]: This code block uses a similar approach to the Price resolver

File: components/price.js [hunk 2]
```diff
@@ -43,39 +43,37 @@ export function PriceProvider ({ price, children }) {
...
+
+const carousel = [
+  'fiat',
+  'yep',
+  '1btc',
+  'blockHeight',
+  'chainFee',
+  'halving'
+]
+
```
Comment (tsmith123) [inline components/price.js:56]: Introduced a carousel concept here as better reflects the idea of clicking through items in the header.

File: components/price.js [hunk 3]
```diff
@@ -43,39 +43,37 @@ export function PriceProvider ({ price, children }) {
...
-      setAsSats('fiat')
-    } else {
-      window.localStorage.setItem('asSats', 'yep')
-      setAsSats('yep')
-    }
+    const nextPos = (pos + 1) % carousel.length
+
+    window.localStorage.setItem(STORAGE_KEY, carousel[nextPos])
+    setAsSats(carousel[nextPos])
+    setPos(nextPos)
```
[the remark this answers is not in the document]
Comment (huumn) [reply components/price.js:77]: Nice this was getting out of hand!

**Comment by huumn** [comment]
Welcome to the team! Do you have a lightning address we can send award sats to?

**Comment by tsmith123** [comment]
@huumn Thanks. Yeah it's `stickymarch60@walletofsatoshi.com`

**Comment by tsmith123** [comment]
Ha ha. No worries. Still learning how things work around here 😆

**Pull request merged**